### PR TITLE
debug: init api for metrics 

### PR DESCRIPTION
**Pull request by AndreMouche**
@hicqu @overvenus PTAL
relate PR in kvproto:https://github.com/pingcap/kvproto/pull/232

### CODE REVIEW

File: src/util/metrics.rs [hunk 1]
```diff
@@ -21,3 +24,49 @@ lazy_static! {
...
+            thread::sleep(interval);
+        })
+        .unwrap();
+
+    Some(handler)
+}
+
+pub fn dump_metrics() -> String {
+    let mut buffer = vec![];
+    let encoder = TextEncoder::new();
```
Comment (siddontang) [inline src/util/metrics.rs]: should support Protobuf Encoder, I worry that Prometheus may send a protobuf content request.

Comment (BusyJay) [reply src/util/metrics.rs]: From [the documentation](https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md#format-variants-comparison), protobuf is not supported anymore after 2.0.

Comment (siddontang) [reply src/util/metrics.rs]: Got it.

**Comment by siddontang** [comment]
we should support this in tikv-ctl.

**Comment by overvenus** [comment]
A friendly ping.

**Comment by AndreMouche** [comment]
@overvenus PTAL

**Comment by AndreMouche** [comment]
@overvenus PTAL

**Pull request merged**